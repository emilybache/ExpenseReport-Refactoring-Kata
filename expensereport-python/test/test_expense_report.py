import io
import sys

from approvaltests import approvals, Options
from approvaltests.scrubbers.date_scrubber import DateScrubber

import expense_report

def createExpense(type, amount):
    expense = expense_report.Expense()
    expense.type = type
    expense.amount = amount
    return expense

def test_expense_report():
    orig_sysout = sys.stdout
    try:
        fake_stdout = io.StringIO()
        sys.stdout = fake_stdout

        expenses = [
            createExpense(expense_report.ExpenseType.DINNER, 5000),
            createExpense(expense_report.ExpenseType.DINNER, 5001),
            createExpense(expense_report.ExpenseType.BREAKFAST, 1000),
            createExpense(expense_report.ExpenseType.BREAKFAST, 1001),
            createExpense(expense_report.ExpenseType.CAR_RENTAL, 200)
        ]
        expense_report.ExpenseReport().print_report(expenses)

        answer = fake_stdout.getvalue()
        approvals.verify(answer, options=Options().add_scrubber(
            DateScrubber.get_scrubber_for("Wed Dec 11 14:59:44 2024")
        ))
    finally:
        sys.stdout = orig_sysout

