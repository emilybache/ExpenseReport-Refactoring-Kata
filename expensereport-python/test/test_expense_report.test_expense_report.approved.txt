Expense Report <date0>
Dinner 	 5000 	  
Dinner 	 5001 	 X
Breakfast 	 1000 	  
Breakfast 	 1001 	 X
Car Rental 	 200 	  
Meals: 12002
Total: 12202
